packages/delegate/src/getBatchingExecutor.ts

```diff
@@ -0,0 +1,71 @@
+import { getOperationAST } from 'graphql';
+
+import isPromise from 'is-promise';
+
+import DataLoader from 'dataloader';
+
+import { ExecutionResult, Request } from '@graphql-tools/utils';
+
+import { SubschemaConfig, ExecutionParams } from './types';
+import { memoize2of3 } from './memoize';
+import { mergeRequests } from './mergeRequests';
+import { splitResult } from './splitResult';
+
+export const getBatchingExecutor = memoize2of3(function (
+  _context: Record<string, any>,
+  _subschemaConfig: SubschemaConfig,
+  executor: ({ document, context, variables, info }: ExecutionParams) => ExecutionResult | Promise<ExecutionResult>
+) {
+  const loader = new DataLoader(createLoadFn(executor));
```

Comment:
An advisable characteristic of `DataLoader` to carry through here would be [maxBatchSize](https://github.com/graphql/dataloader#new-dataloaderbatchloadfn--options). Apollo-Link-Batch-Http [also does this](https://www.npmjs.com/package/apollo-link-batch-http#options). Basically – we want to batch as much as possible until it becomes ill-advised, at which time breaking up the queries is actually preferable.

As a developer, I'd probably want to set this during the `stitchSchemas` call, something like:

```
const gateway = stitchSchemas({
  subschemas: [ ... ],
  mergeTypes: true,
  maxBatchSize: 20
});
```

Normally we set these max batch sizes to be pretty generous, erring on the side of performance. However, having the limit there is just sort of a rational safeguard against a fluke that somehow pumps excessive load into the system all at once.

Reply:
Resolved with additional batchingOptions and nested dataLoaderOptions